src/hotspot/share/prims/jvmtiEnvThreadState.cpp

```diff
@@ -335,6 +296,25 @@ class GetCurrentLocationClosure : public HandshakeClosure {
     }
     _completed = true;
   }
+  void do_vthread(Handle target_h) {
+    if (_target_jt != nullptr) {
+      assert(_target_jt->vthread() == target_h(), "sanity check");
+      do_thread(_target_jt);
+      return;
+    }
+    if (!JvmtiEnvBase::is_vthread_alive(target_h())) {
+      return; // _completed remains false.
+    }
+    ResourceMark rm;
+    javaVFrame *jvf = JvmtiEnvBase::get_vthread_jvf(target_h());
```

Comment:
This method already handles both mounted and unmounted case, so do we need the first conditional above?

Reply:
Good suggestion, thanks.
I was also thinking about it but decided to avoid the risk because of this check in `do_thread()`:
```
    if (!jt->is_exiting() && jt->has_last_Java_frame()) {
```
It can be it is not important to check or I can add an assert for this condition. Let me try and test it first.
This kind of simplification looks as important.